integration_tests/src/main/python/json_test.py

```diff
@@ -380,41 +380,46 @@ def test_from_json_map_fallback():
-        lambda spark : unary_op_df(spark, data_gen) \
-            .select(f.from_json(f.col('a'), schema)),
+        lambda spark : unary_op_df(spark, json_string_gen) \
+            .select(f.from_json('a', schema)),
         conf={"spark.rapids.sql.expression.JsonToStructs": True})
 
-@pytest.mark.parametrize('data_gen', [StringGen(r'{"teacher": "Alice", "student": {"name": "Bob", "age": 20}}')])
-@pytest.mark.parametrize('schema', [StructType([StructField("teacher", StringType())]),
-                                    StructType([StructField("student", StructType([StructField("name", StringType()), \
-                                                                                   StructField("age", IntegerType())]))])])
-def test_from_json_struct_of_struct(data_gen, schema):
+@pytest.mark.parametrize('schema', ['struct<teacher:string>',
+                                    'struct<student:struct<name:string,age:int>>',
+                                    'struct<teacher:string,student:struct<name:string,age:int>>'])
+def test_from_json_struct_of_struct(schema):
+    json_string_gen = StringGen(r'{"teacher": "Alice", "student": {"name": "Bob", "age": 20}}').with_special_pattern('', weight=50)
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark : unary_op_df(spark, data_gen) \
-            .select(f.from_json(f.col('a'), schema)),
+        lambda spark : unary_op_df(spark, json_string_gen) \
+            .select(f.from_json('a', schema)),
         conf={"spark.rapids.sql.expression.JsonToStructs": True})
 
-@pytest.mark.parametrize('data_gen', [StringGen(r'{"teacher": "Alice", "student": \[{"name": "Bob", "class": "junior"},' \
-                                                r'{"name": "Charlie", "class": "freshman"}\]}')])
-@pytest.mark.parametrize('schema', [StructType([StructField("teacher", StringType())]),
-                                    StructType([StructField("student", ArrayType(StructType([StructField("name", StringType()), \
-                                                                                             StructField("class", StringType())])))]),
-                                    StructType([StructField("teacher", StringType()), \
-                                                StructField("student", ArrayType(StructType([StructField("name", StringType()), \
-                                                                                             StructField("class", StringType())])))])])
-def test_from_json_struct_of_list(data_gen, schema):
+@pytest.mark.parametrize('schema', ['struct<teacher:string>',
+                                    'struct<student:array<struct<name:string, class:string>>>',
+                                    'struct<teacher:string,student:array<struct<name:string, class:string>>>'])
+def test_from_json_struct_of_list(schema):
+    json_string_gen = StringGen(r'{"teacher": "Alice", "student": \[{"name": "Bob", "class": "junior"},' \
```

Comment:
do we need escaping inside a raw string?

we could also use a multi-line syntax to help readability
```Python
r''''
{
  "a" : "l",
  "c" : 1
  
}
'''
```

Reply:
Updated tests, thank you!